### ISSUE TITLE
Branch Overview not correctly updated when swichting Repos

### ISSUE THREAD

**Issue by Sdfraga** [larshp/abapGit#2047]
Hi,

There is a problem on the ZCL_ABAPGIT_FACTORY=>GET_BRANCH_OVERVIEW( ).

Let's say you have RepoA and you go to Branch->Overview Feature. You will have the overview of the commits of RepoA in a graphic manner. Now you come back to the abapGit home page and swich to RepoB. If you go again to the Branch->Overview Feature, we still have the history of RepoA and not RepoB.

After some debug, it seems that the issue is related with the singleton on GET_BRANCH_OVERVIEW( ) method. Because Singleton is used, GI_BRANCH_OVERVIEW is bound (and pointing to RepoA) so we will not rebuild GI_BRANCH_OVERVIEW for RepoB.

I would really much like to contribute with the correction but I have never done it, so I would like to ask you for suggestions:

- 1 - Do we really need the singleton for the Branch management? By removing the singleton the issue would be solved but some performance would be lost.

- 2 - If we really want to use Singleton DP, maybe we could implement a [Multiton](https://en.wikipedia.org/wiki/Multiton_pattern), and then get the correct instance by  using an internal table in ZCL_ABAPGIT_FACTORY where we store an index and the correspondent branch instances? Then in GET_BRANCH_OVERVIEW, we would retrieve the correct instance by it's index. Is this too much complexity in the current design??

- 3 - Any other suggestion?

Thanks a lot
Sérgio Fraga

**Comment by larshp** [comment]
Hi,
I think the singleton is not needed, just instantiate it every time. It is not used in any core logic, so performance does not matter. 

![image](https://user-images.githubusercontent.com/5888506/47991902-bdbcbf00-e0eb-11e8-96b3-1aa880ba3bbc.png)

`ZCL_ABAPGIT_INJECTOR` is used for injecting a instance into `ZCL_ABAPGIT_FACTORY`, however this is not implemented for GET_BRANCH_OVERVIEW yet, suggest keeping the member/attribute GI_BRANCH_OVERVIEW.

IF NOT gi_branch_overview IS INITIAL. RETURN it, (it originates from injector) else instantiate new object

**Comment by Sdfraga** [comment]
OK.
I will try to manage this later today after reading the "How to Contribute" help page!

**Comment by larshp** [comment]
Also feel free to join the Slack channel(https://abapgit-slackinviter.herokuapp.com/) if you have any questions 😄

**Comment by Sdfraga** [comment]
Thanks for the Slack suggestion @larshp . In fact I am following the project for about 2 years so I am already on the channel. Never had the opportunity to really contribute until now I guess.

Regarding your suggestions, I did not really understood it since it seems the exact same logic as before but just inverted.

Nevertheless, I forked abapGit repo and removed the singleton from GET_BRANCH_OVERVIEW( ).
The branches are now correctly updated for their correspondent repos.

Let me know if it's OK.
